src/python/bot/startup/run_bot.py

```diff
@@ -125,7 +126,17 @@ def task_loop():
       update_task.track_revision()
       if environment.is_uworker():
         # Batch/Swarming tasks only run one at a time.
-        sys.exit(utasks.uworker_bot_main())
+        uworker_return_code = None
+        exception = None
+        try:
+          uworker_return_code = utasks.uworker_bot_main()
+        except Exception as e:
+          exception = e
+        finally:
+          monitor.stop()
```

Comment:
You might be able to do this more simply with a context manager:

```py
with monitor.Daemon() as monitoring_daemon:
  sys.exit(utasks.uworker_bot_main())
```

Reply:
I never questioned myself as to why this with thing would show up in Python code. Thanks for calling that out!